csharp/autobuilder/Semmle.Autobuild.Shared/BuildScript.cs

```diff
@@ -192,6 +193,25 @@ public override int Run(IBuildActions actions, Action<string, bool> startCallbac
         public static BuildScript Create(Func<IBuildActions, int> func) =>
             new ReturnBuildCommand(func);
 
+        /// <summary>
+        /// Creates a build script that downloads the specified file.
+        /// </summary>
+        public static BuildScript DownloadFile(string address, string fileName) =>
+            Create(actions =>
```

Comment:
Do we already have some retry mechanism anywhere in the autobuilder? It might be worth retrying a failed download (maybe in a followup PR). I used `Polly` before, which was very easy to configure, but then we would have an extra 3rd party dependency.

Reply:
We could build a combinator
```csharp
        public static BuildScript Retry(BuildScript script, int attempts)
        {
            var ret = BuildScript.Failure;
            while (attempts-- > 0)
                ret |= script;
            return ret;
        }
```
but I am not sure if we want to use it for downloading `nuget.exe`.